### DISCUSSION TITLE
How can I wait until the death animation is done?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1060] [Help] [answered]
I added an event listener to remove entities after they die, but this stops the death animation. Is there a "correct" way to wait until the animation is complete before removing the entity?

Here's what I have so far:
```kotlin
handler.addListener(EntityDeathEvent::class.java) { event ->
    event.entity.remove()
}
```

Thanks!

**Comment by GoldenCookieGIT** [comment, marked as answer]
trial and error, wait for x ticks until it feels right for your application then remove the entity

**Reply by ryleu** [reply]
what should i use to wait? i don't want to be blocking by running `Thread.sleep`

**Reply by emortaldev** [reply]
You can use Minestom's scheduler:

```
MinecraftServer.getSchedulerManager().buildTask(() -> {
   // do stuff
}).delay(X, TimeUnit.SERVER_TICK).schedule()
```

Or you can use any sort of non blocking java scheduler. Java schedulers will likely be more accurate (I've noticed 1 tick repeating tasks aren't actually 1 tick)

**Reply by ryleu** [reply]
@emortaldev thanks! it seems like 1 second is perfect. here's what i ended up writing:
```kt
handler.addListener(EntityDeathEvent::class.java) { event ->
    // remove the body after the animation is done
    event.entity.scheduleRemove(Duration.ofSeconds(1))
}
```
and it feels pretty much exactly like vanilla

**Reply by emortaldev** [reply]
schedule remove is lovely too :p

**Comment by VegetalDev** [comment]
Late and possibly irrelevant, but if you're only dealing with `EntityCreature`, you can use its `getRemovalAnimationDelay` method.
